### ISSUE TITLE
The request header is partially missing

### ISSUE THREAD

**Issue by w8350196** [cfug/dio#1547]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | flutter                    |      |
| Platform Version               |                            |      |
| Dio Version                    | 4.0.1                           |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

We use Dio to send the network request  and we will put some data in the request header.
occasionally there is some data loss in the request header acquired by the service side.
Has anyone experienced this problem?
